### ISSUE TITLE
False Negative, regression: `useless_conversion` not reported on `.any()`

### ISSUE THREAD

**Issue by theKidOfArcrania** [rust-lang/rust-clippy#16794]
### Summary

I think this #14800 created too broad of an exception in terms of reporting `useless_conversion` of an `into_iter()` call.

### Lint Name

useless_conversion

### Reproducer

I tried this code:

```rust
use std::iter::IntoIterator;

fn hi(s: &str) -> bool {
    s.chars()
        .into_iter()
        .any(|c| c == 'a')
}

fn hi2(a: u32) -> bool {
    let s: String = format!("hello {a}").into();
    s.chars()
        .into_iter()
        .into_iter()
        .any(|c| c == 'a')
}
```

I expected to see this happen:
```
warning: useless conversion to the same type: `std::str::Chars<'_>`
 --> clippytest/src/main.rs:4:5
  |
4 | /     s.chars()
5 | |         .into_iter()
  | |____________________^ help: consider removing `.into_iter()`: `s.chars()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
  = note: `#[warn(clippy::useless_conversion)]` on by default

warning: useless conversion to the same type: `std::string::String`
  --> clippytest/src/main.rs:10:21
   |
10 |     let s: String = format!("hello {a}").into();
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into()`: `format!("hello {a}")`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion

warning: useless conversion to the same type: `std::str::Chars<'_>`
  --> clippytest/src/main.rs:11:5
   |
11 | /     s.chars()
12 | |         .into_iter()
13 | |         .into_iter()
   | |____________________^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
help: consider removing `.into_iter()`
   |
11 ~     s.chars()
12 +         .into_iter()
   |

warning: `clippytest` (bin "clippytest") generated 3 warnings (run `cargo clippy --fix --bin "clippytest"` to apply 3 suggestions)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.16s
```

Instead, this happened:
```
warning: useless conversion to the same type: `std::string::String`
  --> clippytest/src/main.rs:10:21
   |
10 |     let s: String = format!("hello {a}").into();
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into()`: `format!("hello {a}")`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.94.0/index.html#useless_conversion
   = note: `#[warn(clippy::useless_conversion)]` on by default

warning: `clippytest` (bin "clippytest") generated 1 warning (run `cargo clippy --fix --bin "clippytest" -p clippytest` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
```

### Version

```text
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: x86_64-unknown-linux-gnu
release: 1.94.1
LLVM version: 21.1.8
```
